Bump Lua to v5.4.3

ararslan opened the pull request:
Lua is notorious for not following semver. We'll see whether they've changed the build system...

giordano commented:
> We'll see whether they've changed the build system...

I take it as a "yes"

giordano commented:
Uhm, before merging this we need to fix compat of [dependants](https://juliahub.com/ui/Packages/Lua_jll/Lde6c/5.3.6+0?t=2): the package has a different filename on Windows, that'd break them :disappointed:  Also, can we require Julia v1.6 and build for new platforms?

ararslan commented:
Did I do that right?

ararslan commented on F/flux_core/build_tarballs.jl:
```diff
@@ -71,7 +71,7 @@ dependencies = [
     Dependency("Hwloc_jll"),
     Dependency("SQLite_jll"),
-    Dependency("Lua_jll"),
+    Dependency("Lua_jll"; compat="5.3"),
```
If this uses Pkg's version resolution stuff, should this be
```suggestion
    Dependency("Lua_jll"; compat="~5.3"),
```
since otherwise the implicit `^` means that 5.4 would be acceptable?

giordano commented:
No, don't do it in this pull request, I want to fix the registry, there is no need to trigger a build for the jll packages

ararslan commented:
Can you show an example of the change you're describing? Like submit a PR to the JLL repos so that a new version would add compat to the registry? Or back-compat all existing versions? Or both

giordano commented:
* In Yggdrasil: https://github.com/JuliaPackaging/Yggdrasil/pull/3116 (use `[skip ci] [skip build]` in the commit message, we aren't going to actually rebuild the packages, just prepare the build recipes for the future)
* in the registry: https://github.com/JuliaRegistries/General/pull/37834

In this case I think the compatibility will be `"~5.3.6"`, the name of the library depends on major+minor version number.  Thanks!

giordano commented:
For the record, with #3210 we now have readline for all platforms, including the experimental ones

ararslan commented:
I dropped the commits from this PR that were setting compatibility and moved them to...
- Registry: https://github.com/JuliaRegistries/General/pull/39444
- Yggdrasil: https://github.com/JuliaPackaging/Yggdrasil/pull/3211

I also pushed a commit to try building for experimental platforms.

giordano commented on L/Lua/build_tarballs.jl:
```diff
@@ -31,7 +31,7 @@ fi
 if [[ ${target} == *-mingw* ]]; then
     TO_BIN="lua.exe luac.exe"
-    TO_LIB="lua53.dll"
+    TO_LIB="lua54.dll"
```
I wonder what happens if we call this `liblua.dll`? :slightly_smiling_face:

ararslan replied:
That's a good question. My only concern would be for downstream libraries that make naming assumptions on Windows for this library and wouldn't know to look for `liblua.dll`.

giordano replied:
That's a good point :disappointed:

ararslan replied:
Should probably leave as-is for now, at least.

ararslan commented:
Thanks as always for your patient and thorough help, Mosè 🙂
